### PR TITLE
Add client credentials flow doc

### PR DESCRIPTION
**Pull request by moarychan**
Feature request: 
[Get access token via Client Credential flow](https://github.com/Azure/azure-sdk-for-java/issues/20219)

Related PR: 
[Add client credentials flow for b2c starter](https://github.com/Azure/azure-sdk-for-java/pull/20695)

Changelog:
- Add client credentials flow doc in the aad-b2c-starter readme.
- fix some other module issues.

### CODE REVIEW

**Comment by backwind1233** [comment]
@backwind1233

**Comment by backwind1233** [comment]
@backwind1233

File: sdk/spring/azure-spring-boot-samples/azure-spring-cloud-sample-eventhubs-multibinders/src/main/java/com/azure/spring/sample/eventhubs/multibinders/EventProducerController.java [hunk 1]
```diff
@@ -33,13 +33,13 @@
     public ResponseEntity<String> sendMessage1(@RequestParam String message) {
         LOGGER.info("Going to add message {} to Sinks.Many1", message);
```
Comment (backwind1233) [inline sdk/spring/azure-spring-boot-samples/azure-spring-cloud-sample-eventhubs-multibinders/src/main/java/com/azure/spring/sample/eventhubs/multibinders/EventProducerController.java:34]: how about change the log info , "Sinks.Many1"-->to  the method name "sendMessage1"

File: sdk/spring/azure-spring-boot-samples/azure-spring-cloud-sample-eventhubs-multibinders/src/main/java/com/azure/spring/sample/eventhubs/multibinders/EventProducerController.java [hunk 2]
```diff
@@ -33,13 +33,13 @@
...
         many1.emitNext(MessageBuilder.withPayload(message).build(), Sinks.EmitFailureHandler.FAIL_FAST);
-        return ResponseEntity.ok("Sent1!");
+        return ResponseEntity.ok(message);
     }
 
     @PostMapping("/messages2")
     public ResponseEntity<String> sendMessage2(@RequestParam String message) {
         LOGGER.info("Going to add message {} to Sinks.Many2.", message);
```
Comment (backwind1233) [inline sdk/spring/azure-spring-boot-samples/azure-spring-cloud-sample-eventhubs-multibinders/src/main/java/com/azure/spring/sample/eventhubs/multibinders/EventProducerController.java:41]: same here

**Review by backwind1233 — CHANGES_REQUESTED** [review]

File: sdk/spring/azure-spring-boot-samples/azure-spring-cloud-sample-eventhubs-multibinders/src/main/java/com/azure/spring/sample/eventhubs/multibinders/EventProducerController.java (hunk 1 above)
Comment (moarychan) [reply sdk/spring/azure-spring-boot-samples/azure-spring-cloud-sample-eventhubs-multibinders/src/main/java/com/azure/spring/sample/eventhubs/multibinders/EventProducerController.java:34]: OK

File: sdk/spring/azure-spring-boot/src/samples/java/com/azure/spring/autoconfigure/b2c/ResourceServerConfiguration.java [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+// Licensed under the MIT License.
+
+package com.azure.spring.autoconfigure.b2c;
+
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+
+@EnableWebSecurity
+public class ResourceServerConfiguration extends WebSecurityConfigurerAdapter {
```
Comment (backwind1233) [inline sdk/spring/azure-spring-boot/src/samples/java/com/azure/spring/autoconfigure/b2c/ResourceServerConfiguration.java:11]: @EnableGlobalMethodSecurity

File: sdk/spring/azure-spring-boot/src/samples/java/com/azure/spring/autoconfigure/b2c/WebappAccessResourceController.java [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.reactive.function.client.WebClient;
+
+import static org.springframework.security.oauth2.client.web.reactive.function.client.ServletOAuth2AuthorizedClientExchangeFilterFunction.clientRegistrationId;
+
+@RestController
+public class WebappAccessResourceController {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(WebappAccessResourceController.class);
+    private static final String LOCAL_WEB_API_A_SAMPLE_ENDPOINT = "http://localhost:8081/webApiA/webApiB/sample";
```
Comment (backwind1233) [inline sdk/spring/azure-spring-boot/src/samples/java/com/azure/spring/autoconfigure/b2c/WebappAccessResourceController.java:19]: rename here

**Review by backwind1233 — CHANGES_REQUESTED** [review]

File: sdk/spring/azure-spring-boot/src/samples/java/com/azure/spring/autoconfigure/b2c/ResourceServerController.java [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+    /**
+     * webApiA resource api for web app
+     * @return test content
+     */
+    @PreAuthorize("hasAuthority('APPROLE_WebApiA.SampleScope')")
+    @GetMapping("/webApiA/sample")
+    public String callWebApiASample() {
+        LOGGER.info("WebApiA callSample() returned.");
+        return "Request '/webApiA/sample'(WebApi A) returns success.";
+    }
```
Comment (backwind1233) [inline sdk/spring/azure-spring-boot/src/samples/java/com/azure/spring/autoconfigure/b2c/ResourceServerController.java:34]: @PreAuthorize("hasAuthority('APPROLE_WebApiA.SampleScope')")
    @GetMapping("/webApiA/sample")
    public String callWebApiASample() {
        LOGGER.info("call callWebApiASample()");
        return "Request '/webApiA/sample'(WebApi A) returns successfully.";
    }

**Review by backwind1233 — CHANGES_REQUESTED** [review]

**Review by backwind1233 — APPROVED** [review]
LGTM

**Pull request merged**